authappliance/menu.py

```diff
@@ -967,6 +968,68 @@ def delete(self, tag):
             # Delete backup job.
             self.Audit.del_rotate(None, hour, minute, month, dom, dow)
 
+class UpdatesMenu(object):
+    def __init__(self, app, dialog):
+        self.app = app
+        self.d = dialog
+        self.updates = Updates()
+
+    def menu(self):
+        while True:
+            self.updates.read()
+            choices = [(self.add_update, "Add new update cronjob", "")]
+            for cronjob, options in self.updates.get_update_cronjobs():
+                choices.append((partial(self.delete_update, cronjob),
+                                " ".join(cronjob.time),
+                                "'{}' packages".format(options['-t'])))
+            menu = self.d.value_menu("Updates",
+                                     choices=choices,
+                                     cancel='Back',
+                                     backtitle='Automatic Updates Configuration')
+            if menu is not None:
+                menu()
+            else:
+                break
+
+    def add_update(self):
+        bt = "Add an automatic update job"
+        code, update_date = self.d.inputbox("The date to run the update. "
+                                      "Please enter it like this:\n"
+                                      "<Minute>  <Hour>  <Day-of-Month> "
+                                      " <Month>  <Day-of-Week>\n"
+                                      "You may use '*' as wildcard entry.",
+                                      width=70,
+                                      backtitle=bt)
+
+        if code != self.d.DIALOG_OK:
+            return
+
+        date_fragments = update_date.split()
+        if len(date_fragments) > 5:
+            return
+
+        update_type = self.d.value_radiolist("Please specify which updates should be installed.",
+                                             choices=[
+                                                 (UPDATE_SECURITY, "security", "Install security updates only"),
+                                                 (UPDATE_UPDATES, "updates", "Install general updates")
+                                             ],
+                                             current=UPDATE_SECURITY,
+                                             backtitle=bt)
+        if update_type is None:
+            return
+
+        code = self.d.yesno("Should the system be rebooted after updates have been installed?",
```

Comment:
The system will only be rebooted, if the update requires a reboot, like a new kernel being installed.
If no reboot is necessary, it will not be rebootet. So I am wondering what is the right text here. Maybe we need to write a bit more like:

> "Some update might require a reboot to take effect, like installing a new kernel. Should the system be rebootet after the update, if the update requires a reboot?"